### ISSUE TITLE
add converter function for a specific field

### ISSUE THREAD

**Issue by amanenk** [huandu/go-sqlbuilder#69]
I am using postgresql and want to use json column. I have described json tags and want to make sql builder parse data before insertion into table.
table structure is next
```
CREATE TABLE IF NOT EXISTS json_test
(
    id uuid UNIQUE NOT NULL,
    meta  jsonb
);
```
golang structure below:
```
type JsonTest struct {
	id uuid.UUID `db:"id"`
	Meta []Meta`db:"meta"`
}

type Meta struct {
	Name string `json:"name"`
	Value string `json:"value"`
}
```
Is there a way to set function that Marshals my data to a correct values? If no It would be great to add such functionality.

**Comment by amanenk** [comment]
It was a problem on my side. it works already!

**Comment by tairosonloa** [comment]
How did you solved it, I have the same case (a slice attr I want to store as json) but getting the following error (following your struct names)

`sql: converting argument $5 type: unsupported type []Meta, a slice of struct`

**Comment by tairosonloa** [comment]
After a few hours, I finally solved it. I'll write here the solution for future people coming from Google like me:

So I initially implemented the `[driver.Valuer](https://golang.org/pkg/database/sql/driver/#Valuer) interface` and the `The [sql.Scanner](https://golang.org/pkg/database/sql/#Scanner) interface, such that it unmarshals a JSON byte slice from the database into the struct fields.` as suggested here, but I was getting the above mentioned error:

`sql: converting argument $5 type: unsupported type []Meta, a slice of struct`

So I found the solution in this [Stack Overflow answer](https://stackoverflow.com/questions/58633251/unmarshal-json-array-of-object-obtained-from-postgresql) from `mkopriva` (not the accepted answer).

**TL;DR I was implementing the `sql.Value` interface wrongly. This is how it ended working**

```go
type JsonTest struct {
	id uuid.UUID   `db:"id"`
	Meta MetaSlice `db:"meta"`
}

type MetaSlice []Meta

type Meta struct {
	Name string  `json:"name"`
	Value string `json:"value"`
}

// Make the Attrs struct implement the driver.Value interface. This method
// simply returns the JSON-encoded representation of the struct.
func (ps MetaSlice) Value() (driver.Value, error) {
	return json.Marshal(ps)
}

// Make the Attrs struct implement the sql.Scanner interface. This method
// simply decodes a JSON-encoded value into the struct fields.
func (ps *MetaSlice) Scan(value interface{}) error {
	b, ok := value.([]byte)
	if !ok {
		return errors.New("type assertion to []byte failed")
	}

	return json.Unmarshal(b, &ps)
}
```
